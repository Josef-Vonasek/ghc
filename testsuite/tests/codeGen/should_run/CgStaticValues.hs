{-# LANGUAGE StaticValues #-}
{-# LANGUAGE ImpredicativeTypes #-}

module Main(main) where

import Data.List ( isPrefixOf )
import System.Process

main = readProcess "sh" [] "nm --defined-only CgStaticValues.o"
        >>= const putStr (static g, static (id . (+)), static (id . show))
               . unlines
               . filter isInteresting
               . map (dropWhile (/=' '))
               . lines
  where
    isInteresting :: String -> Bool
    isInteresting x = any (`isPrefixOf` x)
      [ " D Main_g_closure"
      , " D Main_stableZZC0ZZCmainZZCMainZZCstatic_closure"
      , " D Main_stableZZC1ZZCmainZZCMainZZCstatic_closure"
      ]

g = ""

