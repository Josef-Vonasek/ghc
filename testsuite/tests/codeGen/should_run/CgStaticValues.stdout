 D Main_g_closure
 D Main_stableZZC0ZZCmainZZCMainZZCstatic_closure
 D Main_stableZZC1ZZCmainZZCMainZZCstatic_closure
